File: code/references/src/bin/reference-address.rs
// Copyright (c) 2024 Xu Shaohua <shaohua@biofan.org>. All rights reserved.
// Use of this source is governed by General Public License that can be found
// in the LICENSE file.

fn add1() -> i32 {
    let mut x: i32 = 42;
    let x_ref = &mut x;
    *x_ref += 1;
    x
}

fn main() {
    assert_eq!(add1(), 43);
}

File: code/references/src/bin/reference-address.cpp
// Copyright (c) 2024 Xu Shaohua <shaohua@biofan.org>. All rights reserved.
// Use of this source is governed by General Public License that can be found
// in the LICENSE file.

#include <cassert>

int addr1() {
  int x = 42;
  int& x_ptr = x;
  x_ptr += 1;
  return x;
}

int main(void) {
  assert(addr1() == 43);
  return 0;
}
